### PR TITLE
Api/post realestate

### PR DESCRIPTION
**Pull request by pmzpro**
Created POST RealEstate and AddRealEstateRequestDto.

### CODE REVIEW

**Review by jonathan-osorio — APPROVED** [review]

**Review by RafaelMeha — APPROVED** [review]

**Review by Silva096 — CHANGES_REQUESTED** [review]
Post real estate is returning a list instead of just returning 1 RealEstate

File: RealestateApi/Repository/RealEstateRepository.cs
Comment (RafaelMeha) [inline RealestateApi/Repository/RealEstateRepository.cs:1]: Extra line on line 94

File: RealestateApi/Controllers/RealEstateController.cs [hunk 1]
```diff
@@ -25,5 +24,11 @@ public async Task<List<RealEstate>> GetAllRealEstate()
         {
             return await _realEstateService.GetAllRealEstateAsync();
         }
+
+        [HttpPost(Name = "AddRealEstate")]
+        public async Task<RealEstate> AddRealEstateAsync(AddRealEstateRequestDto realEstateData)
```
Comment (Silva096) [inline RealestateApi/Controllers/RealEstateController.cs:29]: instead of "realEstateData" make it "realEstateDto" please.

File: RealestateApi/Repository/Interfaces/IRealEstateRepository.cs [hunk 2]
```diff
@@ -1,10 +1,12 @@
 ﻿using RealEstateApi.Dto.Request;
+using RealEstateApi.Dto.Response;
 using RealEstateApi.Model;
 
 namespace RealEstateApi.Repository.Interfaces
 {
     public interface IRealEstateRepository
     {
         Task<List<RealEstate>> GetAllRealEstateAsync();
+        Task<RealEstate> AddRealEstateAsync(AddRealEstateRequestDto realEstateData);
```
Comment (Silva096) [inline RealestateApi/Repository/Interfaces/IRealEstateRepository.cs:10]: realEstateData -> realEstateDto please.

File: RealestateApi/Service/Interfaces/IRealEstateService.cs [hunk 3]
```diff
@@ -1,9 +1,11 @@
-﻿using RealEstateApi.Model;
+﻿using RealEstateApi.Dto.Request;
+using RealEstateApi.Model;
 
 namespace RealEstateApi.Service.Interfaces
 {
     public interface IRealEstateService
     {
         Task<List<RealEstate>> GetAllRealEstateAsync();
+        Task<RealEstate> AddRealEstateAsync(AddRealEstateRequestDto realEstateData);
```
Comment (Silva096) [inline RealestateApi/Service/Interfaces/IRealEstateService.cs:9]: realestatedata -> realEstateDto

**Review by Silva096 — CHANGES_REQUESTED** [review]
Made some comments in order to improve names and fix the alignment of the code.

File: RealestateApi/Model/RealEstate.cs [hunk 4]
```diff
@@ -10,7 +10,7 @@ public class RealEstate
         public DateTime Build_Date { get; set; }
         public decimal Price { get; set; }     
         public int SquareMeter { get; set; }
-        public string EnergyClass { get; set; } = null!;
+        public string EnergyClass { get; set; }
```
Comment (Silva096) [inline RealestateApi/Model/RealEstate.cs:13]: Why did you delete "= null" ?

Comment (pmzpro) [reply RealestateApi/Model/RealEstate.cs:13]: A mistake made with the last merge I did, and didn't noticed.

File: RealestateApi/Dto/Request/AddRealEstateRequestDto.cs [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+        public int SquareMeter { get; set; }
+        public string EnergyClass { get; set; } = null!;
+        public int CustomerId { get; set; }
+        public int AgentId { get; set; }
+        public int RealEstateTypeId { get; set; }
+        public int CityId { get; set; }
+        public int TypologyId { get; set; }
+    }
+}
+
```
Comment (jonathan-osorio) [inline RealestateApi/Dto/Request/AddRealEstateRequestDto.cs:21]: Extra line not needed

**Review by jonathan-osorio — APPROVED** [review]

**Review by Silva096 — APPROVED** [review]
